fe/fe-core/src/main/java/com/starrocks/privilege/TablePEntryObject.java

```diff
@@ -100,38 +104,41 @@ public static TablePEntryObject generate(GlobalStateMgr mgr, List<String> tokens
-            dbUUID = PrivilegeBuiltinConstants.ALL_DATABASES_UUID;
-            tblUUID = PrivilegeBuiltinConstants.ALL_TABLES_UUID;
-        } else {
-            Database database = mgr.getMetadataMgr().getDb(catalogName, tokens.get(0));
-            if (database == null) {
-                throw new PrivObjNotFoundException("cannot find db: " + tokens.get(0));
-            }
-            dbUUID = database.getUUID();
+            return new TablePEntryObject(
+                    catalogId,
+                    PrivilegeBuiltinConstants.ALL_DATABASES_UUID,
+                    PrivilegeBuiltinConstants.ALL_TABLES_UUID);
+        }
 
-            if (Objects.equals(tokens.get(1), "*")) {
-                tblUUID = PrivilegeBuiltinConstants.ALL_TABLES_UUID;
-            } else {
-                Table table = null;
-                try {
-                    table = mgr.getMetadataMgr().getTable(catalogName, tokens.get(0), tokens.get(1));
-                } catch (StarRocksConnectorException e) {
-                    throw new PrivObjNotFoundException("cannot find table " +
-                            tokens.get(1) + " in db " + tokens.get(0) + ", msg: " + e.getMessage());
-                }
-                if (table == null || table.isOlapView() || table.isMaterializedView()) {
-                    throw new PrivObjNotFoundException("cannot find table " +
-                            tokens.get(1) + " in db " + tokens.get(0));
-                }
-                tblUUID = table.getUUID();
+        String dbUUID = DbPEntryObject.getDatabaseUUID(mgr, catalogName, tokens.get(0));
+        String tableUUID = getTableUUID(mgr, catalogName, tokens.get(0), tokens.get(1));
+        return new TablePEntryObject(catalogId, dbUUID, tableUUID);
+    }
+
+    private static String getTableUUID(GlobalStateMgr mgr, String catalogName, String dbToken, String tableToken)
+            throws PrivObjNotFoundException {
+        checkArgument(!dbToken.equals("*"));
+        if (tableToken.equals("*")) {
+            return PrivilegeBuiltinConstants.ALL_TABLES_UUID;
+        }
+
+        if (CatalogMgr.isInternalCatalog(catalogName)) {
+            Table table;
+            try {
+                table = mgr.getMetadataMgr().getTable(catalogName, dbToken, tableToken);
+            } catch (StarRocksConnectorException e) {
+                throw new PrivObjNotFoundException("cannot find table " +
+                        tableToken + " in db " + dbToken + ", msg: " + e.getMessage());
+            }
+            if (table == null || table.isOlapView() || table.isMaterializedView()) {
+                throw new PrivObjNotFoundException("cannot find table " +
+                        tableToken + " in db " + dbToken);
             }
         }
 
-        return new TablePEntryObject(catalogId, dbUUID, tblUUID);
+        // for table in external catalog, return tableName directly without validation
+        return tableToken;
```

Comment:
this will return tableToken for olap table

Reply:
fixed